### ISSUE TITLE
False positive avoid using redundant else

### ISSUE THREAD

**Issue by RobSlgm** [meziantou/Meziantou.Analyzer#1165]
### Version of the Meziantou.Analyzer NuGet package

3.0.89

### Rule Identifier

MA0071

### Target Framework

net10

### C# Language version

C# 14

### Description

The else-redundant check possibly doesn't consider function calls in the if else condition.

### Reproduction Steps

Minimal code:

```c#
#!/usr/bin/env -S dotnet --

#:package Meziantou.Analyzer@3.0.89
#:property MeziantouAnalysisMode=all-warnings

var sc = new SomeClass();

foreach (var cmp in new string[] { "first", "second", "third" })
{
    if (sc.Verify("first", cmp) is not null)
    {
        Console.WriteLine("Handled");
    }
    else if (sc.Verify("second", cmp) is not null)
    {
        Console.WriteLine("Handled");
        continue;
    }
    else
    {
        Console.WriteLine($"Not handled");
    }
}


sealed class SomeClass
{
    public int? Verify(string tag, string cmp)
    {
        return tag.Equals(cmp, StringComparison.Ordinal) ? 1 : null;
    }
}
```

The expected output is:
Handled
Handled
Not handled

Without an else the output would be:
Handled
**Not handled**
Handled
Not handled
